tools/container/wrapper/wrapper.go

```diff
@@ -0,0 +1,115 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+func main() {
+	program, err := os.Executable()
+	if err != nil {
+		log.Fatalf("failed to get executable: %v", err)
+	}
+	if isRuntimeWrapper(program) && !isNvidiaModuleLoaded() {
+		log.Println("nvidia driver modules are not yet loaded, invoking runc directly")
+		program, err := exec.LookPath("runc")
+		if err != nil {
+			log.Fatalf("failed to find runc: %v", err)
+		}
+		argv := []string{"runc"}
+		argv = append(argv, os.Args[1:]...)
+		execve(program, argv, os.Environ())
+	}
+	argv := makeArgv(program)
+	envv := makeEnvv(program)
+	execve(program+".real", argv, envv)
+}
+
+func isRuntimeWrapper(program string) bool {
+	return filepath.Base(program) == "nvidia-container-runtime" ||
+		filepath.Base(program) == "nvidia-container-runtime.cdi" ||
+		filepath.Base(program) == "nvidia-container-runtime.legacy"
+}
+
+func isNvidiaModuleLoaded() bool {
+	_, err := os.Stat("/proc/driver/nvidia/version")
```

Comment:
What is the motivation for using this over:
```
cat /proc/modules | grep -e \"^nvidia \" >/dev/null 2>&1
```
here?

Reply:
Note that I have also created https://github.com/NVIDIA/nvidia-container-toolkit/pull/764 which *should* pull this logic into the runtime -- guarded by a feature flag which is set by the containerized "installer".

Reply:
> What is the motivation for using this over:
> 
> ```
> cat /proc/modules | grep -e \"^nvidia \" >/dev/null 2>&1
> ```
> 
> here?

Avoid a regular expression.

Reply:
> Note that I have also created #764 which _should_ pull this logic into the runtime -- guarded by a feature flag which is set by the containerized "installer".

Great, we can drop that code.

Reply:
Ah thank you, I didn't appreciate that aspect. Then great, that can be removed.